libvast/vast/concept/hashable/hash_append.hpp

```diff
@@ -329,20 +329,11 @@ void hash_append(Hasher& h,
 
 // -- tuple -------------------------------------------------------------------
 
-namespace detail {
-
-template <class Hasher, class ...T, size_t ...I>
-void tuple_hash(Hasher& h, const std::tuple<T...>& t,
-                std::index_sequence<I...>) noexcept {
-  (hash_append(h, std::get<I>(t)), ...);
-}
-
-} // namespace detail
-
 template <class Hasher, class ...T>
 std::enable_if_t<!detail::is_contiguously_hashable<std::tuple<T...>, Hasher>{}>
 hash_append(Hasher& h, const std::tuple<T...>& t) noexcept {
-  detail::tuple_hash(h, t, std::index_sequence_for<T...>{});
+  std::apply([&h](auto&& x) { hash_append(h, std::forward<decltype(x)>(x)); },
```

Comment:
```suggestion
  std::apply([&h](auto&&... xs) { hash_append(h, std::forward<decltype(xs)>(xs)...); },
```

As-is this only works for single-element tuples. Might need additional text reflow for clang-format, but you get the idea.

Reply:
It sounds like we're lacking a unit test that hashes a tuple with multiple elements. It might be a good idea to add one for complex refactorings like this so we know the output doesn't change.

(Also, wouldn't the suggested fix result in an endless loop? We never take off the head of the param pack)

Reply:
Oops, thanks for spotting the need for the pack.

I'll add some unit tests and fix up the implementation later tonight likely.

Reply:
Thanks! 👍